### DISCUSSION TITLE
Shortcut for accepting inline suggestions line by line

### DISCUSSION THREAD

**Discussion by hchamre** [zed-industries/zed#20497] [Feature Requests]
Hi! I'm working on porting my custom VS Code keymap to Zed, but I'm encountering an issue with one particular shortcut: accepting inline suggestions line by line. Below is a table showing my attempt to map VS Code actions to their corresponding Zed actions:

| VS Code Action                        | Zed Action                         |
|---------------------------------------|------------------------------------|
| `editor.action.inlineSuggest.acceptNextWord` | `editor::AcceptPartialInlineCompletion` |
| `editor.action.inlineSuggest.acceptNextLine` | **???**                           |
| `editor.action.inlineSuggest.commit`       | `editor::AcceptInlineCompletion` |

Does Zed currently support an action for accepting inline suggestions line by line? If not, would it be possible to submit a feature request for this?

**Comment by SomeoneToIgnore** [comment]
There's no any support for this action indeed and a PR could be good.

I think we can do this by keeping the action set as is and extending the `AcceptPartialInlineCompletion` struct to have a field, which tells how to accept the suggestion partially to the next whitespace or line end.
Keeping the default to whitespace option default, we'll have nothing broken and a new thing added on top relatively organically.
+ we need to test this

Example of an action modification: https://github.com/zed-industries/zed/pull/19593

**Reply by hchamre** [reply]
Thanks for the reply, SomeoneToIgnore. I am unable to do a deep dive and create a PR at the moment, but I will start by creating a feature request in the issue tracker.
